include/revng/Pipeline/Location.h

```diff
@@ -0,0 +1,181 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <array>
+#include <optional>
+#include <string>
+#include <string_view>
+#include <type_traits>
+
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/YAMLTraits.h"
+
+#include "revng/ADT/STLExtras.h"
+#include "revng/Support/MetaAddress.h"
+#include "revng/Support/YAMLTraits.h"
+
+namespace pipeline {
+
+namespace concepts {
+
+template<typename Type>
+concept Rank = std::is_same_v<typename Type::RankTag, Type>;
+
+template<typename Type>
+concept RootRank = Rank<Type> && Type::Depth == 0;
```

Comment:
They are in namespace `concepts`. And `concept`s are by definition `bool`.

Or maybe you don't like using namespaces for this?

Reply:
I don't know, I don't a definitive opinion on naming of concepts, but the closest thing we had to concepts where `std::is_...`. In `requires` the `Is` prefix makes sense, while in place of `typename` it doesn't.

Can you look around for a best-practice so we adopt one once and for all?

Reply:
Closing this thread for now since we had an extensive discussion on the mattermost. If there are any active points to still be discussed here, it'd be better to have separate threads for those.